src/pages/workspace/workflows/approvals/WorkspaceWorkflowsApprovalsApproverPage.tsx

```diff
@@ -140,54 +153,64 @@ function WorkspaceWorkflowsApprovalsApproverPageBeta({policy, personalDetails, i
                 shouldShow: true,
             },
         ];
-    }, [debouncedSearchTerm, personalDetails, policy?.employeeList, selectedApproverEmail, translate]);
+    }, [
+        approvalWorkflow?.approvers,
+        approvalWorkflow?.isDefault,
+        approverIndex,
+        debouncedSearchTerm,
+        defaultApprover,
+        personalDetails,
+        policy?.employeeList,
+        selectedApproverEmail,
+        translate,
+    ]);
 
     const nextStep = useCallback(() => {
-        if (!selectedApproverEmail) {
-            return;
+        if (selectedApproverEmail) {
+            const policyMemberEmailsToAccountIDs = PolicyUtils.getMemberAccountIDsForWorkspace(policy?.employeeList);
+            const accountID = Number(policyMemberEmailsToAccountIDs[selectedApproverEmail] ?? '');
+            const {avatar, displayName = selectedApproverEmail} = personalDetails?.[accountID] ?? {};
+            Workflow.setApprovalWorkflowApprover(
+                {
+                    email: selectedApproverEmail,
+                    avatar,
+                    displayName,
+                },
+                approverIndex,
+                route.params.policyID,
+            );
+        } else {
+            Workflow.clearApprovalWorkflowApprover(approverIndex);
         }
 
-        const policyMemberEmailsToAccountIDs = PolicyUtils.getMemberAccountIDsForWorkspace(policy?.employeeList);
-        const accountID = Number(policyMemberEmailsToAccountIDs[selectedApproverEmail] ?? '');
-        const {avatar, displayName = selectedApproverEmail} = personalDetails?.[accountID] ?? {};
-        Workflow.setApprovalWorkflowApprover(
-            {
-                email: selectedApproverEmail,
-                avatar,
-                displayName,
-                isInMultipleWorkflows: false,
-                isCircularReference: false,
-            },
-            approverIndex,
-        );
-
-        const firstApprover = approvalWorkflow?.approvers?.[0]?.email ?? '';
-        if (!approvalWorkflow?.isBeingEdited && firstApprover) {
-            Navigation.navigate(ROUTES.WORKSPACE_WORKFLOWS_APPROVALS_EDIT.getRoute(route.params.policyID, firstApprover));
+        if (approvalWorkflow?.flow === 'create') {
+            Navigation.goBack();
```

Comment:
why do we need to call both goBack and navigate functions?

Reply:
We have two screens to `pop` and one to `push`. Calling `goBack` twice and then regular `navigate` causes glitching because there is a brief moment where there are no screens in RHP. 

Instead, we call `goBack` once, and then we call `navigate` with the `up` option which replaces the current screen.

Reply:
It is connected to [this explanation](https://github.com/Expensify/App/pull/46798#issuecomment-2277368210)